### PR TITLE
Add support for CPU_RAM_SIZE in GB feature

### PR DESCRIPTION
**Pull request by Sneha-at**
# Description

The PR adds support to specify the CPU RAM size for KV cache in GB, currently CPU RAM can be specified in chunks which is not intuitive.

The rest of the description includes relevant details and context, examples:
- why is this change being made,
- the problem being solved and any relevant context,
- why this is a good solution,
- some information about the specific implementation,
- shortcomings of the solution and possible future improvements.

If the change fixes a Github issue, please include a link, e.g.,:
Usability improvement: Allows to set the CPU size and the buffer size in GB, earlier the user could only set them in chunks which is harder to correlate.

# Tests

Please describe how you tested this change, and include any instructions and/or
commands to reproduce.
Verified functionality with following tests 
a. https://github.com/vllm-project/tpu-inference/blob/cpu-offloading/dev-merge/examples/offload/gke/pod_tpu_host_offload_unit_tests.yaml
b. https://github.com/vllm-project/tpu-inference/blob/cpu-offloading/dev-merge/examples/offload/gke/pod_tpu_commons_cpu_offload_verification.yaml
c. https://github.com/vllm-project/tpu-inference/blob/cpu-offloading/dev-merge/examples/offload/gke/pod_tpu_commons_cpu_offload.yaml

Manual testing:
Verified benchmark pods for the following scenarios
1. CPU > KV cache > HBM
2. KV cache < HBM

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tests/offload/tpu_offload_accuracy_test.py [hunk 1]
```diff
@@ -116,14 +117,15 @@ def test_kv_cache_cpu_offloading_accuracy_smaller_then_cpu_ram(
             swap_op_type,
             _skip_precompile,
             decode_save,
-             # The total CPU RAM size = # cpu chunks * cpu_chunk_size. cpu_chunk_size represent the number of tokens can fit into a single CPU RAM chunk.
-            # cpu_chunk_size for llama-3.2-3B(used above in test)= 256 
-            # CPU RAM size = 4*256=1024 tokens
-            "4", # TPU_OFFLOAD_NUM_CPU_CHUNKS
+            # Number of tokens that can be accommodated in CPU RAM = (CPU_RAM_SIZE_GB*GB)/KV cache size per token.
```
Comment (dannawang0221) [inline tests/offload/tpu_offload_accuracy_test.py]: Is `CPU_RAM_SIZE_GB` the same as `TPU_OFFLOAD_CPU_BUF_SIZE_GB`?

File: tests/offload/tpu_offload_accuracy_test.py [hunk 2]
```diff
@@ -116,14 +117,15 @@ def test_kv_cache_cpu_offloading_accuracy_smaller_then_cpu_ram(
...
+            # KV cache size per token for llama-3.2-3B(used above in test)= 256KB
+            # #Tokens = (0.3*GB)/256KB=1024 tokens
```
Comment (dannawang0221) [inline tests/offload/tpu_offload_accuracy_test.py]: Do you mean (0.3 * 1GB) / 256KB?

File: tests/offload/tpu_offload_connector_scheduler_test.py [hunk 3]
```diff
@@ -62,20 +63,27 @@ def scheduler_factory():
     def _scheduler(
         block_size: int = _DEFAULT_BLOCK_SIZE,
         offload_decode_save: int = 0,
-        offload_num_staging_blocks: int = -1,
-        offload_num_cpu_chunks: int = -1,
```
Comment (dannawang0221) [inline tests/offload/tpu_offload_connector_scheduler_test.py:66]: What does offload_num_staging_blocks and offload_num_cpu_chunks equals to -1 mean? Are we trying to test disable the cpu offloading feature here? Or this is testing the corner case with illegal input?

Comment (dannawang0221) [reply tests/offload/tpu_offload_connector_scheduler_test.py:66]: nvm, I see we change the logic below

File: tests/offload/tpu_offload_accuracy_test.py (hunk 2 above)
Comment (Sneha-at) [reply tests/offload/tpu_offload_accuracy_test.py]: Done

File: tests/offload/tpu_offload_accuracy_test.py (hunk 1 above)
Comment (Sneha-at) [reply tests/offload/tpu_offload_accuracy_test.py]: It is the same, updated to use TPU_OFFLOAD_CPU_BUF_SIZE_GB

File: tpu_inference/envs.py [hunk 4]
```diff
@@ -129,9 +128,6 @@ def _get_validated_env() -> str | None:
     # Ray compiled DAG channel type for TPU
     "VLLM_USE_RAY_COMPILED_DAG_CHANNEL_TYPE":
     env_with_choices("VLLM_USE_RAY_COMPILED_DAG_CHANNEL_TYPE", "shm", ["shm"]),
-    # kv offload to dram: the number of concurrent save threads
```
Comment (juncgu) [inline tpu_inference/envs.py:132]: why removing this env var? I think we are still using it.

**Review by juncgu-google — COMMENTED** [review]
please fix the comment; otherwise, LGTM.

**Comment by Sneha-at** [comment]
All the unit tests are passing for these changes https://github.com/vllm-project/tpu-inference/blob/cpu-offloading/dev-merge/examples/offload/gke/pod_tpu_host_offload_unit_tests.yaml

**Comment by Sneha-at** [comment]
>please fix the comment; otherwise, LGTM.

Discussed offline, the env variable `TPU_OFFLOAD_SAVE_THREADS` is duplicated in the envs.py file thus removing redundant occurance
